internal/rbd/nodeserver.go

```diff
@@ -1221,31 +1230,100 @@ func (ns *NodeServer) xfsSupportsReflink() bool {
 	return false
 }
 
+// extractPVNameFromPath extracts PV name from path containing PV name.
+func extractPVNameFromPath(path string) string {
+	str := strings.Split(path, "/pvc-")[1]
+	str = strings.Split(str, "/")[0]
+
+	return "pvc-" + str
+}
+
 // NodeGetVolumeStats returns volume stats.
 func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest,
 ) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
+	var (
+		imgInfo rbdImageMetadataStash
+		err     error
+	)
+	var metadataPath string
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
```

Comment:
Have you seen it happen that `GetStagingTargetPath()` is empty? I wonder if there isn't a nicer way to find the location of the metadata json file